remote_challenge_evaluation/remote_evaluation_util.py

```diff
@@ -0,0 +1,89 @@
+import os
+import requests
+from eval_ai_interface import EvalAI_Interface
+
+
+class RemoteEvaluationUtil:
+    def __init__(self):
+        # TODO: Populate the variables for the evaluation
+        self.auth_token = ""  # Go to EvalAI UI to fetch your auth token
+        self.evalai_api_server = ""  # For staging server, use -- https://staging.eval.ai; For production server, use -- https://eval.ai
+        self.queue_name = ""  # Check Manage Tab of challenge for queue name
+        self.challenge_pk = ""  # Check Manage Tab of challenge for challenge PK
+
+        self.save_dir = "./" # Location where submissions are downloaded
+
+        # Create evalai object
+        self.evalai = EvalAI_Interface(
+            self.auth_token, self.evalai_api_server, self.queue_name, self.challenge_pk
+        )
+
+    def download(self, submission):
+        response = requests.get(submission.input_file.url)
+        submission_file_path = os.path.join(self.save_dir, submission.input_file.name)
+        with open(submission_file_path, "wb") as f:
+            f.write(response.content)
+        return submission_file_path
+
+    # TODO: Write an evaluate method, change default parameters.
+    def evaluate(user_submission_file, phase_codename, challenge_pk, phase_pk, submission_pk, test_annotation_file = None **kwargs):
+        # Run the submission with the input file using your own code and data.
+
+        # Once the submission is done, make an API call to our queue to inform that the message is evaluated
+        # self.update_finished(
+        #     evalai,
+        #     phase_pk,
+        #     submission_pk,
+        #     result='[{"split": "<split-name>", "show_to_participant": true,"accuracies": {"Metric1": 80,"Metric2": 60,"Metric3": 60,"Total": 10}}]'
+        # )
+
+        # If there is a failure, update the status to failed
+        # self.update_failed(
+        #     phase_pk,
+        #     submission_pk,
+        #     submission_error,
+        #     ...
+        # )
+        pass
```

Comment:
Can we simply ask them to raise an exception when the submission fails and handle the submission status update ourselves [after this line](https://github.com/Cloud-CV/EvalAI-Starters/pull/62/files#diff-81b432587f152065115fd9bc1e9f2511186bd3b381a510958be48ca815b68029R32)?

And you can update status to failed if the `.evaluate` method throws a exception otherwise just use the results returned as successful submission metrics.